modules/linear_boltzmann_solvers/response_evaluator/response_evaluator.cc

```diff
@@ -71,11 +77,9 @@ ResponseEvaluator::OptionsBlock()
 void
 ResponseEvaluator::SetOptions(const InputParameters& params)
 {
-  const auto& user_params = params.ParametersAtAssignment();
-
-  if (user_params.Has("buffers"))
+  if (params.IsParameterValid("buffers"))
```

Comment:
IIRC, the `ParametersAtAssignment` was used to only set options that were explicitly set, i.e. not defaulted values. The reason for this was because `SetOptions` can be called after initialization to modify options (i.e. switching between forward and adjoint modes). Does the `IsParameterValid` behavior preserve this functionality?